module/VuFind/src/VuFind/I18n/Translator/TranslatorAwareTrait.php

```diff
@@ -196,9 +196,23 @@ protected function translateString(
         $str,
         $tokens = [],
         $default = null,
+        $conditionToken = null,
```

Comment:
Note that the condition doesn't actually have to be numeric.  This exists, 
> new_user_welcome_text = "Welcome to %%library%%. A new account has been opened for %%firstname%% %%lastname%%. Your username is %%username%%. Please set a password on this page: %%url%%"

so we could add

> new_user_welcome_text_token_matches_maccabee = "No books for you!" 

or whatever